hbase_sql/src/test/java/com/splicemachine/hbase/PhysicalDeletionIT.java

```diff
@@ -171,4 +154,39 @@ public void testSettingMinRetentionPeriodToInvalidMinRetentionPeriodThrows() thr
             Assert.assertEquals("'-400' is not in the valid range 'non-negative number'.", e.getMessage());
         }
     }
+
+    private void shouldContainAfterPurge(String table, int numCells, boolean forcePurge) throws Exception {
+        TestConnection conn = classWatcher.getOrCreateConnection();
+        try (Connection connection = ConnectionFactory.createConnection(HConfiguration.unwrapDelegate())) {
+            methodWatcher.execute(String.format("CALL SYSCS_UTIL.SYSCS_FLUSH_TABLE('%s', '%s')", PhysicalDeletionIT.SCHEMA1, table));
+            methodWatcher.execute(String.format("CALL SYSCS_UTIL.SET_PURGE_DELETED_ROWS('%s', '%s', %b)", PhysicalDeletionIT.SCHEMA1, table, forcePurge));
+            methodWatcher.execute(String.format("CALL SYSCS_UTIL.SYSCS_PERFORM_MAJOR_COMPACTION_ON_TABLE('%s', '%s')", PhysicalDeletionIT.SCHEMA1, table));
+            Scan s = new Scan();
+            long[] conglomId = SpliceAdmin.getConglomNumbers(conn, PhysicalDeletionIT.SCHEMA1, table);
+            TableName hTableName = TableName.valueOf("splice:" + conglomId[0]);
+            Table t = connection.getTable(hTableName);
+            ResultScanner scanner = t.getScanner(s);
+            int count = 0;
+            for (Result rr = scanner.next(); rr != null; rr = scanner.next()) {
+                count++;
+            }
+            Assert.assertEquals(numCells, count);
+        }
+    }
+
+    @Test
+    public void testMinRetentionPeriodKeepsHistory() throws Exception {
+        methodWatcher.execute(String.format("CALL SYSCS_UTIL.SET_MIN_RETENTION_PERIOD('%s','%s', 3600)", SCHEMA1, MRP31));
+        // add some data
+        methodWatcher.executeUpdate(String.format("INSERT INTO %s.%s VALUES (1),(2),(3),(4),(5),(6),(7)", SCHEMA1, MRP31));
+        methodWatcher.executeUpdate(String.format("DELETE FROM %s.%s", SCHEMA1, MRP31));
+        methodWatcher.executeUpdate(String.format("INSERT INTO %s.%s VALUES (10),(20)", SCHEMA1, MRP31));
+        Thread.sleep(2000); // wait for commit markers to be written
+        shouldContainAfterPurge(MRP31, 9, false);
+        methodWatcher.execute(String.format("CALL SYSCS_UTIL.SET_MIN_RETENTION_PERIOD('%s','%s', 0)", SCHEMA1, MRP31));
+        SConfiguration config = HConfiguration.getConfiguration();
+        long updateInterval = config.getTransactionsWatcherUpdateInterval();
+        Thread.sleep(updateInterval * 1000 + 1000);
+        shouldContainAfterPurge(MRP31, 2, false);
```

Comment:
This is testing that flush or major compaction is keeping / purging data.
It would be nice to test compaction and flush separately, since we have different code paths for those.

Reply:
Let's discuss this again.